src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -70,14 +102,28 @@ internal static SafeFreeCredentials AcquireCredentialsHandle(string package, boo
             SecurityBuffer outSecurityBuffer,
             ref ContextFlagsPal contextFlags)
         {
-            throw new PlatformNotSupportedException();
+            // TODO (Issue #3718): The second buffer can contain a channel binding which is not supported 
+            if ((null != inSecurityBufferArray) && (inSecurityBufferArray.Length > 1))
+            {
+                throw new PlatformNotSupportedException(SR.net_nego_channel_binding_not_supported);
+            }
+
+            return EstablishSecurityContext(
+                (SafeFreeNegoCredentials)credentialsHandle,
+                ref securityContext,
+                false,
+                spn,
+                requestedContextFlags,
+                ((inSecurityBufferArray != null) ? inSecurityBufferArray[0] : null),
```

Comment:
I'm not asking about null, I'm asking about empty, i.e. length == 0.  This line assumes that if inSecurityBufferArray is not null, then it has at least one element, and I'm asking whether that's guaranteed.